### PR TITLE
[charts] Make introduction docs pages for each chart

### PR DESCRIPTION
**Pull request by alexfauquette**
- [x] LIne
- [x] Bar
- [x] Scatter

### CODE REVIEW

File: docs/data/charts/lines/lines.md [hunk 1]
```diff
@@ -7,42 +7,48 @@ title: Charts - Lines
...
+The interpolation between data point can be customized by the `curve` property.
+This property expect one of the follwing string value, corresponding to the interpolation method: `'catmullRom'`, `'linear'`, `'monotoneX'`, `'monotoneY'`, `'natural'`, `'step'`, `'stepBefore'`, `'stepAfter'`.
+
+{{"demo": "InterpolationDemo.js", "bg": "inline"}}
+
 ### CSS
 
-Line pots are made of two elements named `LineElement` and `AreaElement` (will come `SymbolElement`).
-Each element can be selected with the CSS class name `.MuiLineElement-root` or `.MuiAreaElement-root`.
+Line pots are made of three elements named `LineElement`, `AreaElement`, and `MarkElement`.
```
Comment (LukasTy) [inline docs/data/charts/lines/lines.md:46]: Maybe you meant `plots` instead? 🤔 
```suggestion
Line plots are made of three elements named `LineElement`, `AreaElement`, and `MarkElement`.
```

File: docs/data/charts/lines/lines.md [hunk 2]
```diff
@@ -7,42 +7,48 @@ title: Charts - Lines
...
+Each element can be selected with the CSS class name `.MuiLineElement-root`, `.MuiAreaElement-root`, or `.MuiMarkElement-root`.
+
 If you want to select the element of a given series, you can use classes `.MuiLineElement-series-<seriesId>` with `<seriesId>` the id of the series you want to customize.
 
-In the next example, we use the following code to customize line style, and the area of the germany GDP.
+In the next example, we use the following code to customize line style, and the area of the Germany GDP.
```
Comment (LukasTy) [inline docs/data/charts/lines/lines.md:51]:
```suggestion
In the next example, the following code is used to customize the line style and the area of Germany's GDP.
```

File: docs/data/charts/scatter/scatter.md [hunk 3]
```diff
@@ -7,21 +7,17 @@ title: Charts - Scatter
...
-> ⚠️ This feature isn't implemented yet. It's coming.
->
-> 👍 Upvote [issue #7922](https://github.com/mui/mui-x/issues/7922) if you want to see it land faster.
->
-> 💬 To have a solution that meets your needs, leave a comment on the [same issue](https://github.com/mui/mui-x/issues/7922).
-> If you already have a use case for this component, or if you are facing a pain-point with your current solution.
+## Basics
 
-## Basic demo
+Scatter charts series should contain a property `data` containing an array of objects.
```
Comment (LukasTy) [inline docs/data/charts/scatter/scatter.md:12]:
```suggestion
Scatter chart series should contain a `data` property containing an array of objects.
```

File: docs/data/charts/scatter/scatter.md [hunk 4]
```diff
@@ -7,21 +7,17 @@ title: Charts - Scatter
...
+Those objects need properties `x`, `y`, `id`.
```
Comment (LukasTy) [inline docs/data/charts/scatter/scatter.md:13]:
```suggestion
Those objects require `x`, `y`, and `id` properties.
```

**Review by LukasTy — COMMENTED** [review]
A nice structured improvement! 👍

File: docs/data/charts/lines/lines.md (hunk 2 above)
Comment (alexfauquette) [reply docs/data/charts/lines/lines.md:51]:
```suggestion
In the next example, each line style is customized with dashes, and marks are removed.
And the area of Germany's GDP gets a custom gradient color.
```

File: docs/data/charts/lines/InterpolationDemo.tsx
Comment (LukasTy) [inline docs/data/charts/lines/InterpolationDemo.tsx:1]: There is some sort of an issue with this demo. 🙈 

https://github.com/mui/mui-x/assets/4941090/3c05b030-eada-4d1c-9708-212f925d5265

File: docs/data/charts/lines/lines.md [hunk 5]
```diff
@@ -7,48 +7,59 @@ title: Charts - Lines
...
 {{"demo": "StackedAreas.js", "bg": "inline"}}
 
+### Stacking strategy 🚧
+
 ## Styling
 
 ### Interpolation
 
+The interpolation between data points can be customized by the `curve` property.
+This property expects one of the following string value, corresponding to the interpolation method: `'catmullRom'`, `'linear'`, `'monotoneX'`, `'monotoneY'`, `'natural'`, `'step'`, `'stepBefore'`, `'stepAfter'`.
```
Comment (LukasTy) [inline docs/data/charts/lines/lines.md:40]:
```suggestion
This property expects one of the following string values, corresponding to the interpolation method: `'catmullRom'`, `'linear'`, `'monotoneX'`, `'monotoneY'`, `'natural'`, `'step'`, `'stepBefore'`, `'stepAfter'`.
```

File: docs/data/charts/lines/lines.md [hunk 6]
```diff
@@ -7,48 +7,59 @@ title: Charts - Lines
...
-Line pots are made of two elements named `LineElement` and `AreaElement` (will come `SymbolElement`).
-Each element can be selected with the CSS class name `.MuiLineElement-root` or `.MuiAreaElement-root`.
+Line plots are made of three elements named `LineElement`, `AreaElement`, and `MarkElement`.
+Each element can be selected with the CSS class name `.MuiLineElement-root`, `.MuiAreaElement-root`, or `.MuiMarkElement-root`.
+
 If you want to select the element of a given series, you can use classes `.MuiLineElement-series-<seriesId>` with `<seriesId>` the id of the series you want to customize.
 
-In the next example, we use the following code to customize line style, and the area of the germany GDP.
+In the next example, each line style is customized with dashes, and marks are removed.
+And the area of Germany's GDP gets a custom gradient color.
```
Comment (LukasTy) [inline docs/data/charts/lines/lines.md:52]:
```suggestion
The area of Germany's GDP also gets a custom gradient color.
```

File: docs/data/charts/lines/InterpolationDemo.tsx
Comment (alexfauquette) [reply docs/data/charts/lines/InterpolationDemo.tsx:1]: Yes, I fixed it in the PR about tooltip. The bug came from edge case when comparing the mousse position with data values.

**Review by LukasTy — APPROVED** [review]
🙌

**Pull request merged**